packages/sequelize-data-provider/src/sequelizeDataProvider.js

```diff
@@ -75,17 +78,27 @@ async function getRelatedEntities(entityName, originalEntityId, relationFieldNam
 }
 
 async function createEntity(entityName, entityToCreate) {
-  const listRelations = [];
-  await asyncEach(Object.keys(entityToCreate), async entityField => {
-    if (isObject(entityToCreate[entityField]) && entityToCreate[entityField].connect) {
-      listRelations.push(...(await handleRelatedConnects(entityName, entityField, entityToCreate)));
-    }
-  });
+  const listRelations = await handleEntityRelationsPreCreate(entityName, entityToCreate);
   const createdEntity = await model(entityName).create(entityToCreate);
   await associateRelations(listRelations, createdEntity);
   return createdEntity.dataValues;
 }
 
+async function createManyEntities(entityName, entitiesToCreate) {
+  const entitiesToCreateListRelations = await async.map(entitiesToCreate, async entityToCreate => {
+    const listRelations = await handleEntityRelationsPreCreate(entityName, entityToCreate, CREATE_MANY);
+    return listRelations;
+  });
+
+  const createdEntities = await model(entityName).bulkCreate(entitiesToCreate);
+  await async.eachOf(createdEntities, async (createdEntity, index) => {
+    const listRelations = entitiesToCreateListRelations[index];
```

Comment:
I had to rely on the order of the entities to map between entity to it's listRelation.

Async.map retains order and as far as I can see in sequelize's code bulkCreate returns the results in order but I am not 100%. Didn't find any docs about it.

The tests pass consistently so that gives some more confidence that order is maintained.

Reply:
as I said on slack I have some doubts about whether we can rely on these things... and if the risk we take here is worth the optimization... at the very least i'd try to implement some safety-measurement by comparing the entityToCreate and the createdEntities values (minus the generated-id), but as the entityToCreate might contains nested operations, this might be non trivial.